### PR TITLE
Added RandomColorShifts transform

### PR DESCRIPTION
**Pull request by Dipet**
Generalized version of `HueSaturationValue`. This transform can work with different color spaces. Also supported `random` color space, when we choose color space randomly.
 
Supported color spaces:
```python
class Colorspace(Enum):
        RANDOM = "random"
        HLS = "HLS"
        HSV = "HSV"
        LAB = "LAB"
        LUV = "LUV"
        YCrCb = "YCrCb"
        XYZ = "XYZ"
        YUV = "YUV"
```

This transform can resolve this issue #698

### CODE REVIEW

**Comment by ternaus** [comment]
If it helps to resolve https://github.com/albumentations-team/albumentations/issues/698 we need a test that shows that for a set of the parameters CollorJitter and RandomColorShift match.

**Comment by Dipet** [comment]
Thanks @ternaus for reminding me that we need a test to compare this transforms. `ColorJitter` works with image in different way.

**Comment by sachit-menon** [comment]
Would changing H and S using HLS colorspace with this (composed with `RandomBrightnessContrast`) correspond to the perturbation `ColorJitter` does? A 1:1 comparison with that augmentation would be very useful for people hoping to migrate from torchvision.

**Comment by Dipet** [comment]
I wrote our version of `ColorJitter` #705

File: albumentations/augmentations/functional.py [hunk 1]
```diff
@@ -418,6 +418,62 @@ def shift_hsv(img, hue_shift, sat_shift, val_shift):
...
+
+        lut = np.arange(0, 256, dtype=np.int16)
+        if is_a:
+            lut = np.mod(lut + shift, limit[1]).astype(np.uint8)
+        else:
+            lut = np.clip(lut + shift, limit[0], limit[1]).astype(np.uint8)
+
+        cv2.LUT(channel, lut, dst=channel)
+
+    img = cv2.merge(img)
```
Comment (BloodAxe) [inline albumentations/augmentations/functional.py:437]: Can we avoid cv2.split/cv2.merge by doing cv2.LUT inplace?  `cv2.LUT(img[..., channel], lut, dst=img[..., channel])` ?

**Review by BloodAxe — APPROVED** [review]

File: albumentations/augmentations/functional.py (hunk 1 above)
Comment (Dipet) [reply albumentations/augmentations/functional.py:437]: This doesn't work, because numpy is creating a new array.

Comment (Dipet) [reply albumentations/augmentations/functional.py:437]: This version is slower.
Current version: 120 FPS
`img[..., i] = cv2.LUT(img[..., i], lut)`: 110 FPS

Comment (Dipet) [reply albumentations/augmentations/functional.py:437]: Version `img[..., i] = lut[img[..., i]]` gives 60 FPS

**Comment by Dipet** [comment]
I need some tests with `XYZ`, `YUV` and `YCrCb` colorspaces

**Review by ternaus — APPROVED** [review]

**Comment by Dipet** [comment]
I close this PR, because at this time it is work different for uint8 and float32 data types

**Pull request closed without merging**